brave/src/main/java/brave/Tracer.java

```diff
@@ -224,16 +227,45 @@ public Span newChild(TraceContext parent) {
    * extracted span IDs. This means the extracted context (if any) is the parent of the span
    * returned.
    *
+   * <p><em>Note:</em> If a context could be extracted from the input, that trace is resumed, not
```

Comment:
cc @jcchavezs subtle, but important. When you are calling something like kafka.poll, you might already be in a trace. Currently, the zipkin model does not have support for multiple parents (or links as defined in census). The impact of this is that when you can extract a span from headers, it won't be allocated to the current trace (as the extracted one is in a different trace).

In many cases, there is no impact as tools like message processors are often entrypoints. However, it is interesting to keep in mind in the future. In census, you can establish a trace+span link from one trace to another, though I've not seen it used in practice. One day, we might revisit this, but not today!